Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Storage/Queries/AllocatorJob.hs

```diff
@@ -49,14 +53,22 @@ getTasksById ids = Esq.findAll $ do
   where_ $ job ^. AllocatorJobId `in_` valList (map (.getId) ids)
   pure job
 
-getReadyTasks :: SchedulerM [AnyJob AllocatorJobType]
-getReadyTasks = do
+getReadyTasks :: Int -> SchedulerM [AnyJob AllocatorJobType]
+getReadyTasks maxShards = do
   now <- getCurrentTime
+  shardVal <- fromIntegral <$> Hedis.incr getShardIdKey
```

Comment:
You don't have to check and reset redis counter. We can keep incrementing and use `counter % maxShards` to get the current shard number.